gcp/volume.go

```diff
@@ -184,6 +195,45 @@ func (c *Client) getVolumeByRegion(region string) ([]volumeResult, error) {
 	return volumes, nil
 }
 
+func (c *Client) findRegionbyID(VolumeID string) (volumeResult, error) {
+
+	baseURL := fmt.Sprintf("-/Volumes")
+
+	statusCode, response, err := c.CallAPIMethod("GET", baseURL, nil)
+	if err != nil {
+		log.Print("ListVolumesByName request failed")
+		return volumeResult{}, err
+	}
+
+	responseError := apiResponseChecker(statusCode, response, "findRegionbyID")
+	if responseError != nil {
+		return volumeResult{}, responseError
+	}
+
+	var result []volumeResult
+	if err := json.Unmarshal(response, &result); err != nil {
+		log.Print("Failed to unmarshall response from findRegionbyID")
+		return volumeResult{}, err
+	}
+
+	var count = 0
+	var resultVolume volumeResult
+	for _, eachVolume := range result {
+		if eachVolume.VolumeID == VolumeID {
+			count = count + 1
+			resultVolume = eachVolume
```

Comment:
I think it would be better to build a list of volumes.   So if we find more than one, we can give more information of why the import failed.  The user could delete older volumes in other regions for instance.

Reply:
Yes, we extend it to show the conflicting volumes. But, as you said, deleting would be the only way for resolution for the user, which is an undesirable solution.